self-core-impl/src/test/java/com/selfxdsd/core/SelfCoreTestCase.java

```diff
@@ -107,5 +111,33 @@ public void closesStorageOnClose() throws Exception {
         Mockito.verify(storage, Mockito.times(1)).close();
     }
 
+    /**
+     * Should verify an authenticated user.
+     */
+    @Test
+    public void authenticatedUser() {
+        final User user = Mockito.mock(User.class);
+        Mockito.when(user.username()).thenReturn("john");
+        Mockito.when(user.email()).thenReturn("john@john.com");
+        final Provider provider = Mockito.mock(Provider.class);
+        Mockito.when(user.provider()).thenReturn(provider);
+        final Projects all = Mockito.mock(Projects.class);
+        Mockito.when(user.projects()).thenReturn(all);
+        final Authenticated authenticated = new Authenticated(user, "tok3n");
+        final Provider prov = new Authenticated(user, "tok3n").provider();
```

Comment:
@ANadrowski You don't need to make a new instance of ``Authenticated`` here, just use the one above :D

Reply:
@amihaiemil I did not understand this. Do you have in mind something like this: `final Provider prov = authenticated.provider();` ?

Reply:
@ANadrowski yes, exactly :D

Reply:
@amihaiemil thanks :)